frontend/resourceadm/components/ResourceDeployEnvCard/ResourceDeployEnvCard.tsx

```diff
@@ -40,42 +32,52 @@ export const ResourceDeployEnvCard = ({
   envName,
   currentEnvVersion,
   newEnvVersion,
+  onClick,
+  loading,
 }: ResourceDeployEnvCardProps): React.ReactNode => {
   const { t } = useTranslation();
 
-  const handleOnClick = () => {
-    console.log('Coming soon...');
-  };
-
   return (
     <div className={classes.cardWrapper}>
-      <Paragraph size='small' className={classes.envName}>
-        <strong>{envName}</strong>
-      </Paragraph>
-      <div className={classes.envWrapper}>
-        <Tag color='neutral' variant='outlined' size='small'>
-          v{currentEnvVersion}
-        </Tag>
-        {newEnvVersion && (
-          <>
-            <div className={classes.arrowWrapper}>
-              <ArrowRightIcon title={t('resourceadm.deploy_card_arrow_icon', { env: envName })} fontSize='1.5rem' />
-            </div>
-            <Tag color='success' variant='outlined' size='small'>
-              v{newEnvVersion}
+      {loading ? (
+        <Spinner title={t('resourceadm.deploy_deploying')}></Spinner>
+      ) : (
+        <>
+          <Paragraph size='small' className={classes.envName}>
+            <strong>{envName}</strong>
+          </Paragraph>
+          <Paragraph size='small' className={classes.versionNumberText}>
+            {t('resourceadm.deploy_version_number_text')}
+          </Paragraph>
+          <div className={classes.envWrapper}>
+            <Tag color='neutral' variant='outlined' size='small'>
+              {currentEnvVersion}
             </Tag>
-          </>
-        )}
-      </div>
-      <div className={classes.buttonWrapper}>
-        <Button
-          aria-disabled={!isDeployPossible}
-          onClick={() => isDeployPossible ? handleOnClick() : undefined}
-          size='small'
-        >
-          {t('resourceadm.deploy_card_publish', { env: envName })}
-        </Button>
-      </div>
+            {newEnvVersion && (
+              <>
+                <div className={classes.arrowWrapper}>
+                  <ArrowRightIcon
+                    title={t('resourceadm.deploy_card_arrow_icon', { env: envName })}
+                    fontSize='1.5rem'
+                  />
+                </div>
+                <Tag color='success' variant='outlined' size='small'>
+                  {newEnvVersion}
+                </Tag>
+              </>
+            )}
+          </div>
+          <div className={classes.buttonWrapper}>
+            <Button
+              aria-disabled={!isDeployPossible}
+              onClick={() => (isDeployPossible ? onClick() : undefined)}
+              size='small'
+            >
+              {t('resourceadm.deploy_card_publish', { env: envName })}
+            </Button>
+          </div>{' '}
```

Comment:
Trenger vi denne her? 😊 
```suggestion
          </div>
```

Reply:
nei se der du, den har sneket seg inn. Fjernet nå :)